pyproject.toml

```diff
@@ -51,6 +51,7 @@ dependencies = [
   "requests",
   "pydantic",
   "transformers[torch]==4.25.1",
+  "numpy<1.24",  # Keep compatibility with latest numba
```

Comment:
Do we need this for haystack base dependencies? I am not sure based on [setup.py in Numba](https://github.com/numba/numba/blob/main/setup.py#L375)

If it's only needed for audio node related dependencies, can we move this to audio dependencies?

Reply:
Hi @mayankjobanputra 

There is one issue, and a PR to add support for Numpy 1.24.

https://github.com/numba/numba/pull/8691
https://github.com/numba/numba/issues/8464

After removing the espnet pin, NumPy could go from 1.22 to 1.24, but Numba would break. So I pinned it.

Reply:
Oh superb... but this is still specific to the audio node right? So if we just move `"numpy<1.24"` under audio dependencies it won't stop Haystack from using newer numpy version and will install appropriate numpy version for audio nodes that needs it.

Reply:
I moved it.

I pinned it because `numba` is used at the Document Store base file (`njit`). It was a safeguard for future bugs that could be introduced if any of the Numba x Numpy issues propagate to Haystack.

Reply:
Ah cool. Thanks @danielbichuetti .